app/controllers/api/v2/courses_controller.rb

```diff
@@ -1,19 +1,25 @@
 module API
   module V2
     class CoursesController < API::V2::ApplicationController
+      before_action :build_provider
+
       def index
-        provider = Provider.find_by!(provider_code: params[:provider_code])
-        authorize provider, :can_list_courses?
+        authorize @provider, :can_list_courses?
         authorize Course
 
-        render jsonapi: provider.courses
+        render jsonapi: @provider.courses
       end
 
       def show
-        course = authorize Course.find_by!(course_code: params[:code])
-
+        course = authorize Course.where(provider: @provider).find_by!(course_code: params[:code].upcase)
```

Comment:
I'm going to guess that this wasn't tested? Sounds like a missing test coverage.

Reply:
Will card it up

Reply:
Could we use something a bit more standard, like:

```
course = @provider.courses.find_by!(course_code params[:code].upcase)
authorize course
```

As we add more actions here, we should expect to use a `before_action` hook like `build_course` to standardise how we build @course, which is why it would make sense to have have the authorize separate.